Fix staticcheck failures for vendor/k8s.io/client-go/rest/*

JaredTan95 opened the pull request:
Signed-off-by: JaredTan95 <jian.tan@daocloud.io>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
Part of  https://github.com/kubernetes/kubernetes/issues/92402

```bash
vendor/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go:482:9: assigning the result of this type assertion to a variable (switch err := err.(type)) could eliminate type assertions in switch cases (S1034)
        vendor/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go:497:8: could eliminate this type assertion
vendor/k8s.io/client-go/plugin/pkg/client/auth/openstack/openstack_stub.go:33:14: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/rest/client_test.go:376:96: the argument is already a string, there's no need to use fmt.Sprintf (S1025)
vendor/k8s.io/client-go/rest/config.go:260:8: should convert c (type TLSClientConfig) to sanitizedTLSClientConfig instead of using struct literal (S1016)
vendor/k8s.io/client-go/rest/config_test.go:283:5: error var fakeDialerError should have name of the form errFoo (ST1012)
vendor/k8s.io/client-go/rest/config_test.go:295:5: error var fakeAuthProviderConfigPersisterError should have name of the form errFoo (ST1012)
vendor/k8s.io/client-go/rest/connection_test.go:65:2: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
        vendor/k8s.io/client-go/rest/connection_test.go:55:3: call to T.Fatalf
vendor/k8s.io/client-go/rest/connection_test.go:112:2: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
        vendor/k8s.io/client-go/rest/connection_test.go:62:3: call to T.Fatalf
vendor/k8s.io/client-go/rest/connection_test.go:145:2: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
        vendor/k8s.io/client-go/rest/connection_test.go:62:3: call to T.Fatalf
vendor/k8s.io/client-go/rest/exec.go:24:2: package "k8s.io/client-go/pkg/apis/clientauthentication" is being imported more than once (ST1019)
        vendor/k8s.io/client-go/rest/exec.go:25:2: other import of "k8s.io/client-go/pkg/apis/clientauthentication"
vendor/k8s.io/client-go/rest/exec_test.go:31:2: package "k8s.io/client-go/pkg/apis/clientauthentication" is being imported more than once (ST1019)
        vendor/k8s.io/client-go/rest/exec_test.go:32:2: other import of "k8s.io/client-go/pkg/apis/clientauthentication"
vendor/k8s.io/client-go/rest/request.go:515:32: possible nil pointer dereference (SA5011)
        vendor/k8s.io/client-go/rest/request.go:519:5: this check suggests that the pointer can be nil
vendor/k8s.io/client-go/rest/request.go:515:32: possible nil pointer dereference (SA5011)
vendor/k8s.io/client-go/rest/request.go:519:5: this check suggests that the pointer can be nil
vendor/k8s.io/client-go/rest/request_test.go:927:6: type errorReader is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:931:22: func errorReader.Read is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:932:22: func errorReader.Close is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1190:5: should use 'return err.Error() == "a container name must be specified for pod kube-dns-v20-mz5cv, choose one of: [kubedns dnsmasq healthz]"' instead of 'if err.Error() == "a container name must be specified for pod kube-dns-v20-mz5cv, choose one of: [kubedns dnsmasq healthz]" { return true }; return false' (S1008)
vendor/k8s.io/client-go/rest/request_test.go:1216:6: type fakeUpgradeConnection is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1218:33: func (*fakeUpgradeConnection).CreateStream is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1221:33: func (*fakeUpgradeConnection).Close is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1224:33: func (*fakeUpgradeConnection).CloseChan is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1227:33: func (*fakeUpgradeConnection).SetIdleTimeout is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1230:6: type fakeUpgradeRoundTripper is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1235:35: func (*fakeUpgradeRoundTripper).RoundTrip is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1246:35: func (*fakeUpgradeRoundTripper).NewConnection is unused (U1000)
vendor/k8s.io/client-go/rest/request_test.go:1335:3: redundant return statement (S1023)
vendor/k8s.io/client-go/rest/urlbackoff_test.go:69:3: var backoffSec is of type time.Duration; don't use unit-specific suffix "Sec" (ST1011)
vendor/k8s.io/client-go/rest/watch/decoder_test.go:54:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
        vendor/k8s.io/client-go/rest/watch/decoder_test.go:57:5: call to T.Fatalf
vendor/k8s.io/client-go/rest/watch/decoder_test.go:70:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
        vendor/k8s.io/client-go/rest/watch/decoder_test.go:73:5: call to T.Fatalf
vendor/k8s.io/client-go/testing/fixture.go:513:2: should use 'return !(!resourceCovers)' instead of 'if !resourceCovers { return false }; return true' (S1008)
vendor/k8s.io/client-go/testing/fixture.go:534:2: should use 'return !(!resourceCovers)' instead of 'if !resourceCovers { return false }; return true' (S1008)
vendor/k8s.io/client-go/testing/fixture.go:555:2: should use 'return !(!resourceCovers)' instead of 'if !resourceCovers { return false }; return true' (S1008)
vendor/k8s.io/client-go/third_party/forked/golang/template/funcs.go:55:5: var builtinFuncs is unused (U1000)
vendor/k8s.io/client-go/third_party/forked/golang/template/funcs.go:80:6: func addFuncs is unused (U1000)
vendor/k8s.io/client-go/third_party/forked/golang/template/funcs.go:99:6: func findFunction is unused (U1000)
vendor/k8s.io/client-go/third_party/forked/golang/template/funcs.go:472:5: should use !strings.ContainsAny(s, `'"&<>`) instead (S1003)
vendor/k8s.io/client-go/tools/cache/delta_fifo.go:432:10: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/tools/cache/fifo.go:266:2: should use 'return f.closed' instead of 'if f.closed { return true }; return false' (S1008)
vendor/k8s.io/client-go/tools/cache/heap.go:307:2: should use 'return h.closed' instead of 'if h.closed { return true }; return false' (S1008)
vendor/k8s.io/client-go/tools/cache/reflector.go:234:23: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/tools/clientcmd/config.go:138:2: should use 'return len(o.LoadingRules.ExplicitPath) > 0' instead of 'if len(o.LoadingRules.ExplicitPath) > 0 { return true }; return false' (S1008)
vendor/k8s.io/client-go/tools/clientcmd/helpers.go:35:12: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/tools/clientcmd/validation.go:237:47: error strings should not end with punctuation or a newline (ST1005)
vendor/k8s.io/client-go/tools/clientcmd/validation.go:267:48: error strings should not end with punctuation or a newline (ST1005)
vendor/k8s.io/client-go/tools/clientcmd/validation.go:275:48: error strings should not end with punctuation or a newline (ST1005)
vendor/k8s.io/client-go/tools/clientcmd/validation_test.go:577:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/tools/clientcmd/validation_test.go:581:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/tools/leaderelection/leaderelection.go:101:15: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/tools/leaderelection/leaderelection.go:101:15: error strings should not end with punctuation or a newline (ST1005)
vendor/k8s.io/client-go/tools/leaderelection/metrics.go:75:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go:143:15: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/tools/remotecommand/remotecommand_test.go:93:2: should use a simple channel send/receive instead of select with a single case (S1000)
vendor/k8s.io/client-go/tools/remotecommand/v1.go:116:17: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/tools/remotecommand/v1.go:120:17: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/tools/watch/retrywatcher.go:81:47: should use make(chan watch.Event) instead (S1019)
vendor/k8s.io/client-go/tools/watch/retrywatcher_test.go:125:35: should use make(chan watch.Event) instead (S1019)
vendor/k8s.io/client-go/transport/round_trippers_test.go:449:23: unnecessary use of fmt.Sprintf (S1039)
vendor/k8s.io/client-go/transport/round_trippers_test.go:464:23: unnecessary use of fmt.Sprintf (S1039)
vendor/k8s.io/client-go/transport/round_trippers_test.go:483:34: unnecessary use of fmt.Sprintf (S1039)
vendor/k8s.io/client-go/util/cert/io.go:33:5: should omit comparison to bool constant, can be simplified to !certReadable (S1002)
vendor/k8s.io/client-go/util/cert/io.go:33:30: should omit comparison to bool constant, can be simplified to !keyReadable (S1002)
vendor/k8s.io/client-go/util/cert/io.go:37:5: should omit comparison to bool constant, can be simplified to !certReadable (S1002)
vendor/k8s.io/client-go/util/cert/io.go:41:5: should omit comparison to bool constant, can be simplified to !keyReadable (S1002)
vendor/k8s.io/client-go/util/certificate/certificate_manager.go:412:27: error strings should not be capitalized (ST1005)
vendor/k8s.io/client-go/util/certificate/csr/csr.go:32:2: package "k8s.io/apimachinery/pkg/api/errors" is being imported more than once (ST1019)
        vendor/k8s.io/client-go/util/certificate/csr/csr.go:33:2: other import of "k8s.io/apimachinery/pkg/api/errors"
vendor/k8s.io/client-go/util/connrotation/connrotation_test.go:103:27: should use time.Since instead of time.Now().Sub (S1012)
vendor/k8s.io/client-go/util/jsonpath/parser.go:355:7: should omit comparison to bool constant, can be simplified to !begin (S1002)
vendor/k8s.io/client-go/util/workqueue/metrics.go:185:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/util/workqueue/metrics.go:189:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/util/workqueue/metrics.go:193:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/util/workqueue/metrics.go:197:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/util/workqueue/metrics.go:201:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/util/workqueue/metrics.go:205:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
vendor/k8s.io/client-go/util/workqueue/metrics.go:209:7: receiver name should not be an underscore, omit the name if it is unused (ST1006)
```

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->


#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

brahmaroutu committed:
Removing utils/mount dependencies from vendor

neolit123 committed:
kubeadm: remove the cgroup driver detection code for Docker

Ideally this should be part of dockershim/CRI and not on the
side of kubeadm.

Remove the detection during:
- During preflight
- During kubelet config defaulting

JornShen committed:
add api e2e test for port ranges of netpolices

jjacobelli committed:
Add last restart time to 'RESTARTS' column to 'kubectl get pods' output

This commit adds the last time since a container has restarted in a pod
to the 'RESTARTS' column to the 'kubectl get pods' output

Signed-off-by: Jordan Jacobelli <jordanjacobelli04@gmail.com>

jjacobelli committed:
Update test cases for 'RESTARTS' column in 'kubectl get pods'

Signed-off-by: Jordan Jacobelli <jordanjacobelli04@gmail.com>

lauchokyip committed:
Fixed sort-by not sorting Resources as expected

pohly committed:
storage e2e: disable health check containers

They are not needed for any of the tests and may be causing too much
overhead (see
https://github.com/kubernetes/kubernetes/issues/102452#issuecomment-854452816).

We already disabled them earlier and then re-enabled them again
because it wasn't clear how much overhead they were causing. A recent
change in how the sidecars get
deployed (https://github.com/kubernetes/kubernetes/pull/102282) seems
to have made the situation worse again. There's no logical explanation
for that yet, though.

(cherry picked from commit 0c2cee5676e64976f9e767f40c4c4750a8eeb11f)

ravisantoshgudimetla committed:
api: Introduce minReadySeconds,AvailableReplicas in Statefulset

The minReadySeconds field on StatefulSet specifies the minimum
number of seconds for which a newly created Pod should be
ready without any of its containers crashing, for it to
be considered available. The AvailableReplicas field
in the status reflects the replicas that are available

ravisantoshgudimetla committed:
validation: Handle presence of api introduced

When the StatefulSetMinReadySeconds feature gate is disabled,
the registry and validation must properly handle dropping the
minReadySeconds and AvailableReplicas fields

wzshiming committed:
Support subresource match

Signed-off-by: Shiming Zhang <wzshiming@foxmail.com>

wzshiming committed:
Add test

Signed-off-by: Shiming Zhang <wzshiming@foxmail.com>

lunhuijie committed:
Add test cases to the LoadClientConfig function

Signed-off-by: 刁浩 10284789 <diao.hao@zte.com.cn>

basantsa1989 committed:
Adding IPV6 (Dual Stack) support to handle IPV6 pod cidrs

Commit:
cleanup tempDir after fsstore_test.go

JaredTan95 commented:
/ok-to-test

caesarxuchao commented:
/assign @roycaihw 
/triage accept

caesarxuchao commented:
/triage accepted

roycaihw commented:
/kind cleanup
/ok-to-test

roycaihw commented on staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go:
```diff
@@ -23,6 +23,7 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"k8s.io/klog"
```
Please remove all the "k8s.io/klog" lines. It's mentioned in the staticcheck result, and we already have "k8s.io/klog/v2"

roycaihw commented on staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go:
```diff
@@ -479,7 +480,7 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 // It must be called while holding the Authenticator's mutex.
 func (a *Authenticator) wrapCmdRunErrorLocked(err error) error {
-	switch err.(type) {
+	switch err := err.(type) {
```
Hmm this is weird. Does L501 still work with this assignment?

roycaihw replied:
I see how it works now. TIL

roycaihw commented on staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go:
```diff
@@ -300,7 +301,7 @@ func (c *commandTokenSource) Token() (*oauth2.Token, error) {
 	cmd.Stderr = &stderr
 	output, err := cmd.Output()
 	if err != nil {
-		return nil, fmt.Errorf("error executing access token command %q: err=%v output=%s stderr=%s", fullCmd, err, output, string(stderr.Bytes()))
```
I don't see this mentioned in the staticcheck report you pasted. Did I miss something?

roycaihw commented on staging/src/k8s.io/client-go/rest/config_test.go:
```diff
@@ -277,10 +277,10 @@ func (n *fakeNegotiatedSerializer) DecoderToVersion(serializer runtime.Decoder,
 }
 
-var fakeDialerError = errors.New("fakedialer")
+var ErrorFakeDialer = errors.New("fakedialer")
```
> error var fakeDialerError should have name of the form errFoo

maybe `errFakeDialer`. A variable starting with a capital letter is public, which is not necessary here

roycaihw commented on staging/src/k8s.io/client-go/rest/watch/decoder_test.go:
```diff
@@ -54,7 +54,7 @@ func TestDecoder(t *testing.T) {
 			data, err := runtime.Encode(scheme.Codecs.LegacyCodec(v1.SchemeGroupVersion), expect)
 			if err != nil {
-				t.Fatalf("Unexpected error %v", err)
+				t.Errorf("Unexpected error %v", err)
```
> vendor/k8s.io/client-go/rest/watch/decoder_test.go:54:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)

this is not the right fix. A proper fix may require surfacing the error back to the main goroutine through a channel.

roycaihw commented on staging/src/k8s.io/client-go/rest/watch/decoder_test.go:
```diff
@@ -70,7 +70,7 @@ func TestDecoder(t *testing.T) {
 			action, got, err := decoder.Decode()
 			if err != nil {
-				t.Fatalf("Unexpected error %v", err)
+				t.Errorf("Unexpected error %v", err)
```
similar here

JaredTan95 commented:
/assign @deads2k

roycaihw commented on staging/src/k8s.io/client-go/rest/connection_test.go:
```diff
@@ -43,6 +43,7 @@ type tcpLB struct {
 }
 
 func (lb *tcpLB) handleConnection(in net.Conn, stopCh chan struct{}) {
+	errs := make(chan error)
```
this is not surfacing the error to the parent goroutine, and will block the send below. I'm afraid a proper fix is more complex than the rest of the PR is: 
- the parent goroutine needs to receive from the channel, to properly call `t.Fatalf`
- we need to be careful about when the goroutines return. If a parent goroutine returns before a child goroutine finishes / sends to the channel, the error will be ignored.
I wonder if we should leave this staticcheck error as it is and fix it in a separate PR.

JaredTan95 replied:
Okay , I will leave this in another PR. But, if that, I don't konw `vendor/k8s.io/client-go/rest` and `vendor/k8s.io/client-go/rest/watch` should be removed from `hack/.staticcheck_failures`?

roycaihw replied:
sadly yes.. but if you could paste the new static check result and show that now it only contains this one failure, I think that's a great achievement.

roycaihw commented on staging/src/k8s.io/client-go/rest/request.go:
```diff
@@ -512,11 +512,15 @@ func (r Request) finalURLTemplate() url.URL {
 	url := r.URL()
 
+	if url == nil {
+		return *url
```
perhaps
```
if url == nil {
    klog.V(4).Infof("unexpected nil URL in Request: %v", r)
    return url.URL{}
}
```

roycaihw commented on staging/src/k8s.io/client-go/rest/request.go:
```diff
@@ -545,7 +549,7 @@ func (r Request) finalURLTemplate() url.URL {
 		// this should not happen that the only two possibilities are /api... and /apis..., just want to put an
 		// outlet here in case more API groups are added in future if ever possible:
 		// https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups
-		// if a wrong API groups name is encountered, return the {prefix} for url.Path
```
`url.Path` looks correct here

JaredTan95 replied:
It's a mis-change.

roycaihw commented on staging/src/k8s.io/client-go/rest/watch/decoder_test.go:
```diff
@@ -93,6 +95,7 @@ func TestDecoder(t *testing.T) {
 		}()
 		<-done
 
+		close(errs)
```
similar here. We should receive from `errs` and properly call `t.Fatalf`

roycaihw reviewed (commented):
I left some comments. LGTM otherwise

roycaihw commented:
/lgtm

Please squash the commits

JaredTan95 commented:
> squash the commits

done

roycaihw commented on staging/src/k8s.io/client-go/rest/watch/decoder_test.go:
```diff
@@ -93,6 +95,7 @@ func TestDecoder(t *testing.T) {
 		}()
 		<-done
 
+		close(errs)
```
this is similar to the other place we discussed before. We cannot close the channel without calling `t.Fatalf`. Better be fixed in a separate PR.

JaredTan95 replied:
got it.

roycaihw reviewed (commented):
one comment. LGTM otherwise

JaredTan95 commented:
/test pull-kubernetes-unit

wojtek-t committed:
Implement support for watch initialization in P&F

JaredTan95 commented:
CC @roycaihw

chenyw1990 committed:
Delete AvailableResources judgment for GA features

jsafrane committed:
Retry detaching FibreChannel volume few times

When UnmountDevice() of a FibreChannel volume fails after unmounting the
device and before the device is fully cleaned up, subsequent
UnmountDevice() retry won't find the device mounted and return without
retrying the device cleanup.

Therefore implement its own retry inside UnmountDevice() to make sure that
the volume devices are either fully cleaned or the error is serius enough
that even 1 minute of trying does not help.

roycaihw commented:
/lgtm

Thanks!

neolit123 committed:
kubeadm: add SkipPhases in {Init|Join}Configuration for v1beta3

- Add the field SkipPhases in the public v1beta3 as a []string (omitempty)
- Add the field in the internal type
- Run generators
- Adapt v1beta2 converter for JoinConfiguration

neolit123 committed:
kubeadm: use "SkipPhases" from configuration for "init" and "join"

During Runner data initialization, if the value for the flag
"--skip-phases" was empty set the {init|join}Runner.Options.SkipPhases
to the {Init|Join}Configuration.SkipPhases value.

neolit123 committed:
kubeadm: add "+optional" to fields with "omitempty" in v1beta3

Add the "+optional" generator tag to all fields in v1beta3 structs
that have "omitempty" in JSON tags.

serathius committed:
Enable protocol-buffers by default

ravisantoshgudimetla committed:
generated: Changes for api introduced

tdihp committed:
Azure auth forwarding adal refresh error to tokenSource, fixes error when token refresh fails.

Signed-off-by: Ping He <tdihp@hotmail.com>

enj commented:
/remove-sig auth

tkashem committed:
client-go: add retry logic for Watch and Stream

damemi committed:
(scheduler e2e) Create balanced pods in parallel

jsafrane committed:
ConsistentRead tries 10 times

We've seen clusters where 3 attempts were not enough. Bumping to 10. The
slowdown should be negligible and it will reduce retry attempts in the
upper layers of kubelet.

jsafrane committed:
Retry reading /proc/mounts indifinetly in FC and iSCSI volume reconstruction

iSCSI and FC volume plugins do not implement real 3rd party attach/detach.
If reconstruction fails with an error on a FC or iSCSI volume, it will not
be unmounted from the volume global dir and at the same time it will be
marked as unused, to be available to be mounted on another node.

The volume can then be mounted on several nodes, resulting in volume
corruption.

The other block based volume plugins implement attach/detach that either
makes the volume stuck (can't be detached) or will be force-detached from a
node before attaching it somewhere else.

aramase committed:
remove LinuxOnly tags for subset of dual-stack tests

Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

timebertt committed:
Fix VolumeAttachment garbage collection for migrated PVs

ravisantoshgudimetla committed:
testdata: StatefulSet

mattcary committed:
Update pd csi driver images to use v1 images (CSINode, CSIDriver, etc)

Change-Id: I51ddf0014dcfe476495e324ee33aa658e7199477

MadhavJivrajani committed:
change log line to print actual pod uid instead of address of the pod uid

Signed-off-by: Madhav Jivrajani <madhav.jiv@gmail.com>

ahg-g committed:
use ownerReference to build default spreading contraints in the scheduler

jsafrane committed:
Bump k8s.io/utils

To get io/ConsistentRead updates.

gjkim42 committed:
kubelet: Validate the length of the DNS search path

gjkim42 committed:
Add feature gate ExpandedDNSConfig

ExpandedDNSConfig allows kubernetes to have expanded DNS(Domain Name
System) configuration

gjkim42 committed:
AllowExpandedDNSConfig if haveSameExpandedDNSConfig(newPod, oldPod)

ravisantoshgudimetla committed:
feature: Add minReadySeconds featuregate

Commit:
Part of master to controlplane in test/integration(1.22)

Commit:
remove scheduler deprecated algorithm-provider flag and clean up algorithmprovider pkg

Commit:
Part3: master to controlplane in test/integration
Rename RunAMaster to RunAControlPlane

serathius committed:
Update etcd makefile to build v3.5.0-beta.3 image

Commit:
part of master to controlplane in test/integration
SetMaster -> SetAPIServer

Commit:
Part of master to controlplane in test/integration
NewMasterConfig -> NewControlPlaneConfig

pohly committed:
storage e2e: use csi-driver-host-path v1.7.2 in single pod

The new default deployment in that release puts sidecars into the same
pod as the driver. This is expected to reduce load during testing.

dims committed:
[scheduler] avoid comparing function pointers in unit tests

PluginFactory is a function that returns a plugin. We have been
comparing these functions in unit tests and it has worked so far, but
starts to fail in gotip/master.

Note from the golang team:
```
Func values are incomparable. It is true that you could get the PC
through reflection but it is still not expected to be compared. PCs
can change due to inlining, wrappers, etc., depending on the
compiler's details, which is subject to change.
```

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

bart0sh committed:
e2e: hugepages: delete test pod after the test

Current test assumes that test pod is deleted when the test
namespace is deleted. However, namespace deletion is an asynchronous
operation. The pod may still be running and allocating hugepages
resources when next test case creates another pod that requests
the same hugepages resources. This can cause kubelet to fail the test
pod with this kind of error:
  OutOfhugepages-2Mi: Node didn't have enough resource: hugepages-2Mi
  requested: 6291456, used: 6291456, capacity: 10485760

Explicitly deleting test pod should fix this issue.

jayunit100 committed:
Implement a windows Netpol NewModel

tsmetana committed:
Local PV e2e: fix leaked local volumes

k8s-ci-robot committed:
Merge pull request #101992 from zshihang/bound

BoundServiceAccountTokenVolume ga

k8s-ci-robot committed:
Merge pull request #102288 from dims/avoid-checking-func-pointers

[scheduler] avoid comparing function pointers in unit tests

k8s-ci-robot committed:
Merge pull request #102291 from bart0sh/PR0096-e2e-hugepages-delete-pod

e2e: hugepages: delete test pod after the test

k8s-ci-robot committed:
Merge pull request #102067 from tkashem/apf-always-create-suggested

apf: always create missing bootstrap configuration object

julianvmodesto committed:
Make a public ValidateAnnotationsSize

liggitt committed:
Grant PSP permissions to all serviceaccounts in e2e, not just default

xmudrii committed:
Update debian-iptables to buster-v1.6.1

Signed-off-by: Marko Mudrinić <mudrinic.mare@gmail.com>

xmudrii committed:
Update debian-base to buster-v1.7.0

Signed-off-by: Marko Mudrinić <mudrinic.mare@gmail.com>

k8s-ci-robot committed:
Merge pull request #101154 from basantsa1989/dual-stack

Adding IPV6 (Dual Stack) support to handle IPV6 pod cidrs

k8s-ci-robot committed:
Merge pull request #102277 from serathius/etcd-image

Update etcd makefile to build v3.5.0-beta.3 image

xmudrii committed:
Update etcd image revision

Signed-off-by: Marko Mudrinić <mudrinic.mare@gmail.com>

k8s-ci-robot committed:
Merge pull request #102302 from xmudrii/debian-iptables-update

Update the Debian images to pick up CVE fixes in the base images

k8s-ci-robot committed:
Merge pull request #101853 from sanposhiho/scheduler_perf/fix/error-handling-correctly

scheduler_perf: correct error handling

k8s-ci-robot committed:
Merge pull request #102252 from smarterclayton/pod_termination

e2e: Flag questionable durations in container status on termination

ahg-g committed:
Return UnschedulableAndUnresolvable instead of Error when failing to lookup pvc or storageclass in VolumeZone plugin

k8s-ci-robot committed:
Merge pull request #102150 from julianvmodesto/ssa-big-last-applied-followup

Make apimachineryvalidation.totalAnnotationSizeLimitB public

k8s-ci-robot committed:
Merge pull request #101935 from arkbriar/kube-proxy/fix-checkout-changes-for-endpoint-slice-cache

kube-proxy: Fix EndpointSliceCache::getEndpointsMap for different endpoints with same IP

k8s-ci-robot committed:
Merge pull request #102236 from ahg-g/ahg-spread

Use ownerReference to build default spreading constraints

k8s-ci-robot committed:
Merge pull request #100394 from mengjiao-liu/clean-up-redundant-code

Fix use nil error

navist2020 committed:
kubeadm:return error info normally

saschagrunert committed:
Build non-static binaries with PIE buildmode

We now add the `-buildmode pie` flag when building non-static binaries,
which enables the ASLR security mechanism.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

lentzi90 committed:
Update CNI plugins v0.9.1

ref: https://github.com/containernetworking/plugins/releases/tag/v0.9.1
Signed-off-by: Lennart Jern <lennart.jern@est.tech>

k8s-ci-robot committed:
Merge pull request #102086 from mengjiao-liu/fix-data-race

Prevent data race condition in vsphere unit tests

k8s-ci-robot committed:
Merge pull request #102301 from liggitt/psp-sa-group-e2e

Grant PSP permissions to all serviceaccounts in e2e, not just default

k8s-ci-robot committed:
Merge pull request #101308 from pacoxu/doc-kubelet-running-pods

kubelet_running_pods shows number of pods that have a running pod sandbox

k8s-ci-robot committed:
Merge pull request #102234 from sanposhiho/scheduler/add/interface-check-on-nodeaffinity

scheduler/add: interface check on nodeaffinity

pacoxu committed:
upgrade klog to v2.9.0

Signed-off-by: pacoxu <paco.xu@daocloud.io>

SataQiu committed:
cleanup PodPreset testdata

Commit:
Promote cronjob to batch/v1: kubectl sparse_test prune

pacoxu committed:
depracate ValidateProxyRedirects as it is with StreamingProxyRedirects

Signed-off-by: pacoxu <paco.xu@daocloud.io>

k8s-ci-robot committed:
Merge pull request #98077 from jayunit100/jay-netpol-win

NetworkPolicy E2E enablement for Windows

k8s-ci-robot committed:
Merge pull request #102269 from lunhuijie/optimizationReturnPlus

simplify returning boolean expression in /pkg/volume

k8s-ci-robot committed:
Merge pull request #102332 from pacoxu/klog-2.9.0

Bump klog to 2.9.0, fixing byte array display

k8s-ci-robot committed:
Merge pull request #100435 from lauchokyip/fixSort

Fixed sort-by not sorting Resources as expected

seans3 committed:
Changes headers to IETF standard

jsafrane committed:
Move error reporting to volume plugins

Move reporting of GetReliableMountRefs error to the volume plugins that
have more context about severity of the error.

k8s-ci-robot committed:
Merge pull request #100531 from sjenning/storage-e2e-reduce-pod-count

e2e: reduce pods used in shared local pv test

k8s-ci-robot committed:
Merge pull request #96130 from brahmaroutu/mount-utils-cleanup

Removing utils/mount dependencies from vendor

k8s-ci-robot committed:
Merge pull request #100842 from ravisantoshgudimetla/add-minReadySeconds-ss

api: Add min ready seconds & availablereplicas for statefulsets

k8s-ci-robot committed:
Merge pull request #101647 from pacoxu/disable-StreamingProxyRedirects

Default StreamingProxyRedirects to disabled

k8s-ci-robot committed:
Merge pull request #102200 from sanwishe/simplifyboolexpr

simplify return boolean value expression

k8s-ci-robot committed:
Merge pull request #102212 from mattcary/csidriver

Update pd csi driver images to use v1 images (CSINode, CSIDriver, etc)

k8s-ci-robot committed:
Merge pull request #102282 from pohly/csi-driver-update

storage e2e: use csi-driver-host-path v1.7.2 in single pod

k8s-ci-robot committed:
Merge pull request #100746 from mengjiao-liu/fix-nil-call

Fixed a possible nil pointer dereference caused by variable `plug`

k8s-ci-robot committed:
Merge pull request #100651 from gjkim42/feature-gate-expanded-dns-config

Add feature gate ExpandedDNSConfig

k8s-ci-robot committed:
Merge pull request #99777 from JornShen/add_e2e_api_validate_endport

add API e2e test for port ranges of netpolices

lunhuijie committed:
Log spelling formatting and a redundant conversion

Signed-off-by: 刁浩 10284789 <diao.hao@zte.com.cn>

k8s-ci-robot committed:
Merge pull request #102239 from Haleygo/clean-up-AlgorithmProvider-flag-and-pkg

clean up algorithmprovider pkg and remove scheduler deprecated algorithm-provider flag

knight42 committed:
fix: skip pods with empty ip

Commit:
CronJob: switch storage version to batch/v1 in 1.22

saschagrunert committed:
Update kube-cross to v1.16.4-2

This patch bumps kube-cross to the latest release.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

k8s-ci-robot committed:
Merge pull request #101960 from knight42/fix/deflake-metrics-proxy

Deflake tests that need to grab metrics from controller-manager or scheduler

saschagrunert committed:
Simplify kube-cross dependency handling

We can indirectly retrieve the kube-cross version from the
`build/build-image/cross/VERSION` for the sample-apiserver. This allows
us to simplify the handling in `build/dependencies.yaml` as well as
the required approval (via `OWNERS`) if the kube-cross version changes.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

wojtek-t committed:
Address watch panics in P&F handler and extend testing.

k8s-ci-robot committed:
Merge pull request #102364 from saschagrunert/cross

Update kube-cross to v1.16.4-2

dashpole committed:
fix prometheus-to-sd image for fluentbit

danwinship committed:
Clarify and split up the "not actually SCTP" SCTP NetworkPolicy test

These tests *do* apply to plugins that don't support SCTP.

k8s-ci-robot committed:
Merge pull request #101862 from jsafrane/fix-fc-detach-retry

Retry detaching FibreChannel volume few times

k8s-ci-robot committed:
Merge pull request #102310 from navist2020/fix/NoErrReturn

kubeadm:return error info correctly

k8s-ci-robot committed:
Merge pull request #102343 from seans3/remove-x-headers

Changes headers to remove "X-" prefix (IETF deprecation)

k8s-ci-robot committed:
Merge pull request #102371 from saschagrunert/kube-cross-version

Simplify kube-cross dependency handling

xmudrii committed:
Update setcap to buster-v2.0.1 and add setcap to dependencies.yaml

Signed-off-by: Marko Mudrinić <mudrinic.mare@gmail.com>

k8s-ci-robot committed:
Merge pull request #102363 from mengjiao-liu/promote-cronjob-batchv1-storage

CronJob: switch storage version to batch/v1 in 1.22

k8s-ci-robot committed:
Merge pull request #102336 from SataQiu/remove-podpreset

cleanup PodPreset testdata

prameshj committed:
Add a test for verifying compute basepath.

Compute API changes could modify this, we need a test to detect if it happens.

k8s-ci-robot committed:
Merge pull request #102377 from xmudrii/setcap-buster-v2.0.1

Update setcap to buster-v2.0.1 and add setcap to dependencies.yaml

k8s-ci-robot committed:
Merge pull request #102138 from damemi/balance-pods-parallel

(scheduler e2e) Create balanced pods in parallel

k8s-ci-robot committed:
Merge pull request #102372 from dashpole/missed_images

Fix prometheus-to-sd image for fluentbit

k8s-ci-robot committed:
Merge pull request #102272 from mengjiao-liu/term-NewIntegrationTestMasterConfig

Part2: master to controlplane in test/integration(1.22)

k8s-ci-robot committed:
Merge pull request #102306 from ahg-g/ahg-vol-restrictions

Return UnschedulableAndUnresolvable instead of Error when failing to lookup volume-related resources

m14815 committed:
Fix a grammar error in doc

m14815 committed:
Error message should not be capitalized.

m14815 committed:
Uniform output string.

k8s-ci-robot committed:
Merge pull request #102268 from sanwishe/loggingformat1

cleanup: Optimization logging format for pkg/kubelet

dims committed:
Bump golang.org/x/text to v0.3.6

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

k8s-ci-robot committed:
Merge pull request #101923 from neolit123/1.22-kubeadm-add-skip-phases-v1beta3

kubeadm: add support for skipping phases in v1beta3

andrewsykim committed:
apis: remove Service topologyKeys

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
apis: update generated code after removing Service topologyKeys

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
features: remove alpha ServiceTopology feature gate

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
kube-proxy: remove ServiceTopology implementation

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

k8s-ci-robot committed:
Merge pull request #101925 from neolit123/1.22-kubeadm-add-optional-fields

kubeadm: add "+optional" to fields with "omitempty" in v1beta3

k8s-ci-robot committed:
Merge pull request #102176 from timebertt/fix/volumeattachment-migrated-volumes

Fix VolumeAttachment garbage collection for migrated PVs

k8s-ci-robot committed:
Merge pull request #102328 from lentzi90/update-cni-plugins

Update CNI plugins v0.9.1

andrewsykim committed:
core/v1: add unit tests to ensure deprecated protobuf field numbers are not re-used

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

k8s-ci-robot committed:
Merge pull request #102417 from m14815/commit-21.5.5

Uniform output string.

Commit:
sched: fix Dump's doc comment

k8s-ci-robot committed:
Merge pull request #102039 from serathius/protobuff

Enable protocol-buffers by default

k8s-ci-robot committed:
Merge pull request #102323 from saschagrunert/pie

Build non-static binaries with PIE buildmode

k8s-ci-robot committed:
Merge pull request #99647 from neolit123/1.21-remove-cgroup-driver-detection

kubeadm: remove the cgroup driver detection for docker

k8s-ci-robot committed:
Merge pull request #102292 from tsmetana/fix-local-pv-e2e-pv-leak

Local PV e2e: fix leaked local volumes

goodluckbot committed:
Improve func ToSelectableFields performance for event

k8s-ci-robot committed:
Merge pull request #102432 from llhuii/fix-scheduler-doc-comment

sched: fix Dump's doc comment

serathius committed:
Add tests to logs

Co-Authored-By: mengjiao.liu <mengjiao.liu@daocloud.io>

serathius committed:
Refactor logs options

neolit123 committed:
kubeadm: add utilities to manage users and groups

In the Alpha stage of the feature in kubeadm to support
a rootless control plane, the allocation and assignment of
UID/GIDs to containers in the static pods will be automated.
This automation will require management of users and groups
in /etc/passwd and /etc/group.

The tools on Linux for user/group management are inconsistent
and non-standardized. It also requires us to include a number of
more dependencies in the DEB/RPMs, while complicating the UX for
non-package manager users.

The format of /etc/passwd and /etc/group is standardized.
Add code for managing (adding and deleting) a set of managed
users and groups in these files.

ahg-g committed:
Return UnschedulableAndUnresolvable when looking up volume-related resources returns NotFound error

k8s-ci-robot committed:
Merge pull request #102462 from serathius/logs-test

Add tests to logs

k8s-ci-robot committed:
Merge pull request #102470 from serathius/logs-refactor

Refactor logs options

yuzhiquan committed:
fix volume failing test

k8s-ci-robot committed:
Merge pull request #99886 from lunhuijie/run-test

Fix the url 'version's comparation of github.com/Azure/go-autorest/' lost

k8s-ci-robot committed:
Merge pull request #102359 from lunhuijie/StructuredLog2

Log spelling formatting and a redundant conversion

k8s-ci-robot committed:
Merge pull request #102107 from tkashem/client-go-retry

client-go: add retry logic for Watch and Stream

giuseppe committed:
vendor: upgrade cadvisor to 0.39.2

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

alculquicondor committed:
Cleanup Framework interface

remove repeated methods

xiaoanyunfei committed:
fix watch_cache_capacity metrics

k8s-ci-robot committed:
Merge pull request #102223 from lunhuijie/run-test6

api link is missing

saschagrunert committed:
Fix regression for timed-out stream cleanups

If a stream is already timed-out, then either the data or error stream
may be `nil`. This would cause a segmentation fault, which is now
covered with this patch.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

k8s-ci-robot committed:
Merge pull request #102279 from mengjiao-liu/update-NewMasterConfig

Part5: master to controlplane in test/integration(1.22)

vinayakankugoyal committed:
Add user and group name constants for kubeadm rootless control-plane.

k8s-ci-robot committed:
Merge pull request #102409 from dims/bump-golang.org/x/text-to-v0.3.6

Bump golang.org/x/text to v0.3.6

k8s-ci-robot committed:
Merge pull request #102465 from ahg-g/ahg-vol-restrictions

Return UnschedulableAndUnresolvable when looking up volume-related resource returns NotFound error

kikimo committed:
clean code

kikimo committed:
reuse fake topology manager

k8s-ci-robot committed:
Merge pull request #102232 from MadhavJivrajani/mirror-client-log-line-fix

Change log line to print actual pod uid and not address of the pod uid

ykakarap committed:
use subpath for coredns only for default repository

thockin committed:
Add KEP links to sig-net feature gates

Adjust 1 removal and comment another.

k8s-ci-robot committed:
Merge pull request #102218 from kolyshkin/cgroup-cleanups

pkg/kubelet/cm: cgroup-related cleanups

k8s-ci-robot committed:
Merge pull request #101559 from njuptlzf/fsstore_test

Clean up tempDir after fsstore_test.go is executed

k8s-ci-robot committed:
Merge pull request #102381 from prameshj/basepathtest

Add a test for verifying compute basepath.

k8s-ci-robot committed:
Merge pull request #102494 from vinayakankugoyal/kubeadm-constants

Add user and group name constants for kubeadm rootless control-plane.

k8s-ci-robot committed:
Merge pull request #102497 from aramase/dual-stack-windows

remove LinuxOnly tags for subset of dual-stack tests

k8s-ci-robot committed:
Merge pull request #102502 from ykakarap/kubeadm-coredns-path-change

kubeadm: use subpath for coredns only for default repository

danwinship committed:
Fix spurious Feature tags on some NetworkPolicy tests

The "[Feature:SCTP]" tag was needed on "should not allow access by TCP
when a policy specifies only SCTP" back when SCTP was alpha, because
it wasn't possible to create a policy that even mentioned SCTP without
enabling the feature gate. This no longer applies, and the tag was
removed from the original copy of network_policy.go, but accidentally
got left behind in the netpol/ version.

Likewise, the newly-added "should not allow access by TCP when a
policy specifies only UDP" got tagged "[Feature:UDP]", but this was
never necessary, and is inconsistent with other UDP tests anyway.

Similarly, we need "[Feature:SCTPConnectivity]" on tests that make
SCTP connections, because that functionality is not available in all
clusters, but "[Feature:UDPConnectivity]" is unnecessary and
inconsistent.

cheftako committed:
Default GCE testing to using konnectivity

Turning GRPC based Konnectivity testing on id advance of
removing the SSHTunnel code.
https://testgrid.k8s.io/sig-api-machinery-network-proxy#ci-kubernetes-e2e-gci-gce-network-proxy-grpc&width=25
shows the feature to be stable and it should remain that way.

kikimo committed:
clean code

k8s-ci-robot committed:
Merge pull request #102510 from cheftako/defaultKonn

Default GCE testing to using konnectivity

k8s-ci-robot committed:
Merge pull request #101605 from njuptlzf/flexvloume_test

cleanup: delete tempDir correctly after flexvloume_test is executed

k8s-ci-robot committed:
Merge pull request #102063 from tdihp/fix/azure-auth-refresh

Azure auth forwarding adal refresh error to azureTokenSource

nilo19 committed:
serviceOwnsFrontendIP shouldn't report error when the public IP doesn't match

chendave committed:
Cleanup redundant failure reason in InterPodAffinity plugin

Both `ErrReasonAffinityRulesNotMatch` and `ErrReasonAntiAffinityRulesNotMatch` are
more precise than `ErrReasonAffinityNotMatch`.

Signed-off-by: Dave Chen <dave.chen@arm.com>

k8s-ci-robot committed:
Merge pull request #100558 from yangjunmyfm192085/run-test31

Optimize the structured Logging migration.

k8s-ci-robot committed:
Merge pull request #102201 from chendave/graduate

Graduate prefer nominated node to beta

k8s-ci-robot committed:
Merge pull request #102483 from giuseppe/update-cadvisor

vendor: upgrade cadvisor to 0.39.2

k8s-ci-robot committed:
Merge pull request #102504 from thockin/master

Add KEP links to sig-net feature gates

k8s-ci-robot committed:
Merge pull request #102485 from alculquicondor/cleanup_interface

Cleanup Framework interface

wojtek-t committed:
Fix watch rejections in P&F filter

pohly committed:
storage e2e: capture driver pod changes, including all events

As seen in https://github.com/kubernetes/kubernetes/issues/102452, we
currently don't have pod events for the CSI driver pods because of the
different namespace and would need them to determine whether the
driver gets evicted.

Previously, only changes of the pods where logged. Perhaps even more
interesting are events in the namespace.

k8s-ci-robot committed:
Merge pull request #102221 from kikimo/add-hint-to-fake-topology-manager

Add hint to fake topology manager.

aojea committed:
e2e lb use same variable for endpoint path

aojea committed:
PokeHTTP default timeout to avoid hanging connections

soltysh committed:
Clean apply's prune and kube addons with batch/v1 CronJob

soltysh committed:
Promote CronJobControllerV2 flag to GA

k8s-ci-robot committed:
Merge pull request #102338 from mengjiao-liu/update-cronjob-to-batchv1-kubectl

Promote cronjob to batch/v1: kubectl sparse_test file

k8s-ci-robot committed:
Merge pull request #102484 from xiaoanyunfei/fix/watch_cache_metrics

fix watch_cache_capacity  metrics

k8s-ci-robot committed:
Merge pull request #102171 from wojtek-t/pf_watch_initialization_support

Implement support for watch initialization in P&F

k8s-ci-robot committed:
Merge pull request #102461 from goodluckbot/apiserver-event-toSelectableFields

Improve ToSelectableFields' performance for events

k8s-ci-robot committed:
Merge pull request #102528 from soltysh/cronjob_ga

Clean apply's prune and kube addons with batch/v1 CronJob

aojea committed:
add loadbalancer name to ESIPP tests so regex pick them

k8s-ci-robot committed:
Merge pull request #102509 from danwinship/sctp-is-not-tcp

Clarify and split up the "not actually SCTP" SCTP NetworkPolicy test

k8s-ci-robot committed:
Merge pull request #102518 from chendave/cleanup_constat

Cleanup redundant failure reason in InterPodAffinity plugin

k8s-ci-robot committed:
Merge pull request #100142 from Ethyling/enh-get-lastrestart

Add `LAST RESTART` column to `kubectl get pods`

liggitt committed:
Fix closing of decorated watcher channel on timeout

k8s-ci-robot committed:
Merge pull request #100679 from hdp617/master

doc: remove description that `socat` is required

Commit:
loadbalancer outer poll loops should have a longer timeout that inner loop

k8s-ci-robot committed:
Merge pull request #101633 from llhuii/kubelet/remove-redundant-code

kubelet_pods.go: clean makeEnvironmentVariables

k8s-ci-robot committed:
Merge pull request #101959 from lunhuijie/run-test5

Add test cases to the LoadClientConfig function

k8s-ci-robot committed:
Merge pull request #101982 from hantmac/fix/need-handle-error

handle Unmarshal error

jeremyje committed:
GCE Windows: Only create NPD kubeconfig when the file path is defined.

k8s-ci-robot committed:
Merge pull request #102516 from nilo19/bug/cherry-pick-649

serviceOwnsFrontendIP shouldn't report error when the public IP doesn't match

k8s-ci-robot committed:
Merge pull request #102526 from pohly/csi-driver-pod-events

storage e2e: capture driver pod events

k8s-ci-robot committed:
Merge pull request #102539 from liggitt/websocket-service-watch

Fix closing of decorated watcher channel on timeout

k8s-ci-robot committed:
Merge pull request #102544 from jeremyje/fixnpd

GCE Windows: Only create NPD kubeconfig when the file path is defined.

k8s-ci-robot committed:
Merge pull request #102529 from soltysh/cronjob_fflag

Promote CronJobControllerV2 flag to GA

k8s-ci-robot committed:
Merge pull request #102276 from mengjiao-liu/update-term-RunAMaster

Part3: master to controlplane in test/integration(1.22)

dims committed:
Updating to prometheus/common v0.26.0

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

dims committed:
make lint-dependencies happy!

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

k8s-ci-robot committed:
Merge pull request #102561 from dims/updating-to-prometheus/common-latest-version

Updating to prometheus/common v0.26.0

k8s-ci-robot committed:
Merge pull request #102270 from mengjiao-liu/update-term-control-plane

Part1: master to controlplane in test/integration(1.22)

k8s-ci-robot committed:
Merge pull request #102278 from mengjiao-liu/rename-SetMaster

part4: master to controlplane in test/integration(1.22)

JonZeolla committed:
Remove -k from toCurl output

k8s-ci-robot committed:
Merge pull request #101860 from chenyw1990/deleteAvailableResourceJudgment

Delete AvailableResources judgment for GA features

k8s-ci-robot committed:
Merge pull request #101738 from matthyx/deflake-startupprobe

fix manual trigger of readinessProbe on startupProbe success

k8s-ci-robot committed:
Merge pull request #102572 from JonZeolla/remove-insecure-curl-suggestion

Remove -k from toCurl output

k8s-ci-robot committed:
Merge pull request #102421 from m14815/commit-21.5.1

Fix a grammar error in doc

andrewsykim committed:
api: update API compatibility tests to remove topologyKeys from Service

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

k8s-ci-robot committed:
Merge pull request #102420 from m14815/commit-21.5.2

Error message should not be capitalized.

k8s-ci-robot committed:
Merge pull request #102477 from yuzhiquan/master

FIx failing tests: [sig-storage] Test_ADC_VolumeAttachmentRecovery/Deleted_Pod_with_migrated_PV

k8s-ci-robot committed:
Merge pull request #102533 from aojea/lb_flakes

Lb flakes

k8s-ci-robot committed:
Merge pull request #101082 from jsafrane/translate-aws-allowAutoIOPSPerGBIncrease

Add allowAutoIOPSPerGBIncrease to translated AWS EBS StorageClasses

k8s-ci-robot committed:
Merge pull request #102059 from jsafrane/fix-consistentread

Retry reading /proc/mounts when unable to get a consistent read

k8s-ci-robot committed:
Merge pull request #102489 from saschagrunert/http-stream-nil

Fix regression for timed-out stream cleanups

liggitt committed:
Revert "client-go: add retry logic for Watch and Stream"

This reverts commit 607d3819498e64d969407c3d7cbbb8f53d98f0d4.

pacoxu committed:
upgrade gopkg.in/evanphx/json-patch to v4.11.0

Signed-off-by: pacoxu <paco.xu@daocloud.io>

pacoxu committed:
add ut for preventing dropping null from arrays

Signed-off-by: pacoxu <paco.xu@daocloud.io>

liggitt committed:
Fix race in attachdetach tests

k8s-ci-robot committed:
Merge pull request #99575 from navist2020/fix/logs_info

Simply modify the Kubectl logs information

BenTheElder committed:
update debian-iptables to v1.6.2

picks up https://github.com/kubernetes/release/pull/2106

k8s-ci-robot committed:
Merge pull request #102467 from pacoxu/json-patch-5.5.0

upgrade json-patch to v4.11.0

k8s-ci-robot committed:
Merge pull request #102581 from liggitt/revert-watch-retry

Revert "client-go: add retry logic for Watch and Stream"

k8s-ci-robot committed:
Merge pull request #100939 from wzshiming/feat/subresource-match

Feature support FakeClientset match subresource

CaoDonghui123 committed:
remove redundant code

k8s-ci-robot committed:
Merge pull request #102584 from liggitt/attachdetach-race

Fix race in attachdetach tests

k8s-ci-robot committed:
Merge pull request #102590 from BenTheElder/deb-ipt-script

update debian-iptables to v1.6.2

k8s-ci-robot committed:
Merge pull request #102591 from pohly/disable-health-monitor

storage e2e: disable health check containers

k8s-ci-robot committed:
Merge pull request #102463 from neolit123/1.22-add-usergroup-utils

kubeadm: add utilities to manage users and groups

k8s-ci-robot committed:
Merge pull request #102593 from CaoDonghui123/fix7

remove redundant code

k8s-ci-robot committed:
Merge pull request #102412 from andrewsykim/kill-service-topology

Remove deprecated alpha Service topologyKeys field

gnufied committed:
The test was not very useful and required elevated access

k8s-ci-robot committed:
Merge pull request #102608 from gnufied/remove-test-on-unknown-block-devices

Remove block device metrics unit test

JaredTan95 committed:
rebase

Signed-off-by: JaredTan95 <jian.tan@daocloud.io>

JaredTan95 committed:
Merge branch 'staticcheck_vendor' of https://github.com/JaredTan95/kubernetes into staticcheck_vendor

Signed-off-by: JaredTan95 <jian.tan@daocloud.io>

# Conflicts:
#	staging/src/k8s.io/client-go/rest/request.go
#	staging/src/k8s.io/client-go/transport/round_trippers_test.go

JaredTan95 commented:
/test pull-kubernetes-node-crio-e2e

neolit123 commented:
/remove-area kubeadm
/remove-sig cluster-lifecycle

please update the PR to only include the areas mentioned in the description.

BenTheElder commented:
this PR needs a rebase, the actual changes are obscured by the PR showing up as having more than 700 commits.

roycaihw commented:
let me know when you clean up the commits and I will re-tag the PR
